configmap/informed_watcher.go

```diff
@@ -47,17 +52,46 @@ func NewInformedWatcherFromFactory(sif informers.SharedInformerFactory, namespac
 	}
 }
 
-// NewInformedWatcher watches a Kubernetes namespace for configmap changes.
-func NewInformedWatcher(kc kubernetes.Interface, namespace string) *InformedWatcher {
+// NewInformedWatcher watches a Kubernetes namespace for ConfigMap changes.
+// Optional label requirements allow restricting the list of ConfigMap objects
+// that is tracked by the underlying Informer.
+func NewInformedWatcher(kc kubernetes.Interface, namespace string, lr ...labels.Requirement) *InformedWatcher {
 	return NewInformedWatcherFromFactory(informers.NewSharedInformerFactoryWithOptions(
 		kc,
 		// We noticed that we're getting updates all the time anyway, due to the
 		// watches being terminated and re-spawned.
 		0,
 		informers.WithNamespace(namespace),
+		informers.WithTweakListOptions(addLabelRequirementsToListOptions(lr)),
 	), namespace)
 }
 
+// addLabelRequirementsToListOptions returns a function which injects label
+// requirements to existing metav1.ListOptions.
+func addLabelRequirementsToListOptions(lr []labels.Requirement) internalinterfaces.TweakListOptionsFunc {
+	if lr == nil {
+		return nil
+	}
+
+	return func(lo *metav1.ListOptions) {
+		sel, err := labels.Parse(lo.LabelSelector)
+		if err != nil {
+			panic(fmt.Sprintf("could not parse label selector %q: %w", lo.LabelSelector, err))
+		}
+		lo.LabelSelector = sel.Add(lr...).String()
+	}
+}
+
+// FilterConfigByLabelExists returns an "exists" label requirement for the
+// given label key.
+func FilterConfigByLabelExists(labelKey string) *labels.Requirement {
+	req, err := labels.NewRequirement(labelKey, selection.Exists, nil)
+	if err != nil {
+		panic(fmt.Sprintf("could not construct label requirement: %w", err))
```

Comment:
Return the error and log.Fatal in sharedmain?

Reply:
Right, that's probably a better approach. on it.